source/server/listener_impl.cc

```diff
@@ -348,6 +348,19 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
               validation_visitor_, parent_.server_.api(), parent_.server_.options())),
       quic_stat_names_(parent_.quicStatNames()) {
 
+  if (address_->type() == Network::Address::Type::Ip &&
+      config.address().socket_address().ipv4_compat() &&
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.strict_check_on_ipv4_compat")) {
+    if (address_->ip()->version() == Network::Address::IpVersion::v6) {
+      if (!address_->ip()->ipv6()->v4CompatibleAddress().ok()) {
+        throw EnvoyException("Only the Ipv6 any address and Ipv4-mapped Ipv6 address can be set "
```

Comment:
I think it may be clearer to say `::` and `0.0.0.0` instead of `any`

Reply:
`IPv6` and `IPv4` (capitalization) for consistency with other docs/messages.

Reply:
`Only IPv6 address '::' or valid IPv4-mapped IPv6 addresses can set ipv4_compat`

Reply:
got it.